sktalk/corpus/parsing/cha.py

```diff
@@ -12,22 +12,37 @@ def _extract_metadata(self):
         return self._pla_reader().headers()[0]
 
     def _extract_utterances(self):
-        reader = self._pla_reader().utterances(by_files=False)
-        return [self._to_utterance(read_utterance) for read_utterance in reader]
+        with open(self._path, encoding="utf-8") as f:
+            lines = f.read().splitlines()
+        utterance_info = [self._extract_info(
+            line) for line in lines if not line.startswith("@")]
+
+        # collect all utterance info in a terrible, terrible loop
+        collection = []
+        timing, participant, utterance = None, None, None
+        for info in utterance_info:
+            if info["utterance"] is not None:
+                timing = info["time"]
+                utterance = info["utterance"]
+            else:
+                continue
+            if info["participant"] is not None:
+                participant = info["participant"]
+            complete_utterance = Utterance(
+                participant=participant,
+                time=timing,
+                utterance=utterance)
+            collection.append(complete_utterance)
+        return collection
 
     @classmethod
-    def _to_utterance(cls, read_utterance) -> Utterance:
-        """Convert pylangacq Utterance to sktalk Utterance"""
-        participant = read_utterance.participant
-        time = read_utterance.time_marks
-        time = list(time) if isinstance(time, (list, tuple)) else None
-        text = str(read_utterance.tiers)
-        text = cls._clean_utterance(text)
-        return Utterance(
-            participant=participant,
-            time=time,
-            utterance=text,
-        )
+    def _extract_info(cls, line):
+        participant = cls._extract_participant(line)
+        time = cls._extract_timing(line)
+        utterance = cls._extract_utterance(line)
```

Comment:
Should be a `@staticmethod`

```suggestion
    def _extract_info(line):
        participant = ChaParser._extract_participant(line)
        time = ChaParser._extract_timing(line)
        utterance = ChaParser._extract_utterance(line)
```